### PR TITLE
timg: add libdeflate for head build

### PR DESCRIPTION
**Pull request by speedy-beaver**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/timg.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ class Timg < Formula
   depends_on "ffmpeg"
   depends_on "graphicsmagick"
   depends_on "jpeg-turbo"
+  depends_on "libdeflate"
```
Comment (carlocab) [inline Formula/timg.rb:25]: Why?

Comment (speedy-beaver) [reply Formula/timg.rb:25]: timg is adding libdeflate in HEAD with [this commit](https://github.com/hzeller/timg/commit/818e47cdc16493bbb04fb88e2a33e377c95c132d)

Comment (carlocab) [reply Formula/timg.rb:25]: Then we should restrict this dependency to `head` until it lands in a stable release. There's no reason to force everyone to install `libdeflate` as a dependency while it's unused.

**Review by carlocab — CHANGES_REQUESTED** [review]
The dependency should be restricted to `head` for now.

**Review by carlocab — APPROVED** [review]

**Pull request merged**